infer/environment.cc

```diff
@@ -1187,6 +1187,12 @@ Environment::processBinding(core::Context ctx, const cfg::CFG &inWhat, cfg::Bind
                             fun = lit.asName();
                         }
                     }
+                    if (fun == core::Names::callWithSplat() && core::isa_type<core::NamedLiteralType>(args[1]->type)) {
+                        auto lit = core::cast_type_nonnull<core::NamedLiteralType>(args[1]->type);
+                        if (lit.derivesFrom(ctx, core::Symbols::Symbol())) {
```

Comment:
fwiw, I think we have a faster way to compute this via `literalKind`. Not sure if `literalTypeKind` predates this code or what, but we probably don't need to do a full on ancestor check here for what could be an int comparison.

Reply:
This is a good callout, I will fix that up in a future PR.